src/middlewares.rs

```diff
@@ -18,30 +21,72 @@ pub struct UserState {
 
 pub type CookieUserState = Extension<UserState>;
 
+impl<'a> TryInto<Cookie<'a>> for UserState {
+    type Error = AppError;
+
+    fn try_into(self) -> Result<Cookie<'a>, Self::Error> {
+        let new_state_json = serde_json::to_string(&self).map_err(|err| {
+            dbg!(err);
```

Comment:
```suggestion
            debug!(?err);
```

Reply:
I ended up switching the trait to from `TryInto` -> `Into`